components/match_table/commons/match_table.lua

```diff
@@ -0,0 +1,753 @@
+
+	local opponents = self:_readOpponents(mode)
+
+	self.config = {
+		mode = mode,
+		limit = tonumber(args.limit),
+		displayGameIcons = Logic.readBool(args.gameIcons),
+		showResult = Logic.nilOr(Logic.readBoolOrNil(args.showResult), true),
+		timeRange = self:readTimeRange(),
+		aliases = self:readAliases(mode),
+		vs = {},
+		title = args.title,
+		showTier = not Logic.readBool(args.hide_tier),
+		showIcon = not Logic.readBool(args.hide_icon),
+		showVod = Logic.readBool(args.vod),
+		showStats = Logic.nilOr(Logic.readBoolOrNil(args.stats), true),
+		showOpponent = Logic.nilOr(Logic.readBoolOrNil(args.showOpponent), #opponents > 1 or mode == PLAYER_MODE),
+	}
+
+	Array.forEach(opponents, function(opponent)
+		Table.mergeInto(self.config.aliases, self:getOpponentAliases(mode, opponent))
+	end)
+
+	local vsMode = args.vsMode or mode
+	assert(vsMode == PLAYER_MODE or vsMode == TEAM_MODE, 'Unsupported "|vsMode=" input')
+
+	Array.forEach(self:_readVsOpponents(mode), function(opponent)
+		Table.mergeInto(self.config.vs, self:getOpponentAliases(mode, opponent))
+	end)
+
+	return self
+end
+
+---@param mode MatchTableMode
+---@return standardOpponent[]
+function MatchTable:_readOpponents(mode)
+	local base = mode == PLAYER_MODE and 'player' or 'team'
+	local inputs = self:_readOpponentInputsFromBase(base)
+
+	if Logic.isEmpty(inputs) then
+		assert(self.title.namespace == 0, 'Required ' .. base .. '= argument')
+		table.insert(inputs, self.title.rootText)
+	end
+
+	return Array.map(inputs, function(input) return self:_readOpponent(mode, input) end)
+end
+
+---@param mode MatchTableMode
+---@return standardOpponent[]
+function MatchTable:_readVsOpponents(mode)
+	local inputs = self:_readOpponentInputsFromBase('vs' .. (mode == PLAYER_MODE and 'player' or 'team'))
+
+	return Array.map(inputs, function(input) return self:_readOpponent(mode, input) end)
+end
+
+---@param base string
+---@return string[]
+function MatchTable:_readOpponentInputsFromBase(base)
+	local inputs = Array.extractValues(Table.filterByKey(self.args, function(key)
+		return key:find('^' .. base .. '%d*$') ~= nil
+	end))
+
+	if Logic.isNotEmpty(inputs) or Logic.isEmpty(self.args[base .. 's']) then return inputs end
+
+	return Array.map(mw.text.split(self.args[base .. 's'], ',', true), String.trim)
+end
+
+---@param mode MatchTableMode
+---@param input string
+---@return standardOpponent
+function MatchTable:_readOpponent(mode, input)
+	if mode == PLAYER_MODE then
+		local player = {displayName = input}
+		PlayerExt.populatePageName(player)
+		return {type = 'solo', players = {player}}
+	end
+
+	return {type = 'team', template = input:lower():gsub('_', ' ')}
+end
+
+---@param mode MatchTableMode
+---@return string[]
+function MatchTable:readAliases(mode)
+	local aliases = {}
+	Array.mapIndexes(function(aliasIndex)
+		if not self.args['alias' .. aliasIndex] then return end
+		local alias = self.args['alias' .. aliasIndex]:gsub(' ', '_')
```

Comment:
For ResultsTable we allow CSV input into `args.aliases`, should we use the same format here?

Reply:
imo stick to one input method (since the alias stuff is new)

Reply:
I personally prefer the CSV version we have in ResultsTable then.